### ISSUE TITLE
Error in Xcode 10 beta: Cycle in dependencies between targets 'MyApp' and 'MyLib'; building could produce unreliable results.

### ISSUE THREAD

**Issue by hhfa008** [CocoaPods/CocoaPods#7833]
Using CocoaPods  1.5.3
Using Xcode 10 beta

Issue:

error: Cycle in dependencies between targets 'MyApp' and 'MyLib'; building could produce unreliable results.
Cycle path: MyApp → Pods-MyApp → MyLib → MyApp
Cycle details:
→ Target 'MyApp' has compile command for Swift source files
→ Target 'Pods-MyApp' has target dependency on Target 'MyLib'
→ Target 'MyLib' has copy command from '/Path/Target Support Files/MyLib/MyLib-umbrella.h' to '/Path/MyLib/MyLib.framework/Headers/MyLib-umbrella.h'
○ Target 'MyLib' has link command with output /Path/MyApp/DerivedData/MyApp/Build/Intermediates.noindex/Pods.build/iphonesimulator/MyLib.build/Objects-normal/x86_64/MyLib'
○ Target 'MyLib' has a command with output '//Path/MyLib/MyLib.framework/MyLib'
Showing All Messages

:-1: Cycle in dependencies between targets 'MyApp' and 'MyLib'; building could produce unreliable results.
Cycle path: MyApp → Pods-MyApp → MyLib → MyApp
Cycle details:
→ Target 'MyApp' has compile command for Swift source files
→ Target 'Pods-MyApp' has target dependency on Target 'MyLib'
→ Target 'MyLib' has copy command from '/Path/Target Support Files/MyLib/MyLib-umbrella.h' to '//Path/MyLib/MyLib.framework/Headers/MyLib-umbrella.h'
○ Target 'MyLib' has link command with output /Path/MyApp/DerivedData/MyApp/Build/Intermediates.noindex/Pods.build/iphonesimulator/MyLib.build/Objects-normal/x86_64/MyLib'
○ Target 'MyLib' has a command with output '/Path/MyLib/MyLib.framework/MyLib'

**Comment by warpling** [comment]
Also running into this with CocoaLumberjack

Cocoapods 1.5.3
Xcode 10 beta 2

**Comment by dnkoutso** [comment]
@warpling thanks for the potential workaround. I am going to continue tagging this with waiting input from the original author.

**Comment by ileitch** [comment]
From the Xcode 10 beta 2 release notes Known Issues:
```
Targets with Copy Headers build phases ordered after Compile Sources build phases may fail to
build and emit a diagnostic regarding build cycles. (39880168)

Workaround: Always order Copy Headers build phases before Compile Sources build phases to
work around this issue.
```

Possibly related?

**Comment by igor-makarov** [comment]
@ileitch I have tried reordering the build phases using a post-build script, and started getting another dependency cycle error.

**Comment by glaurent** [comment]
Had this with RxCocoa as well (same reference to the umbrella header). The error wouldn't always be there in Xcode 10 beta 2, but it is with beta 3.

**Comment by kballard** [comment]
We're hitting this in our app with Xcode 10 beta 4.

**Comment by tom-wolters** [comment]
Clearing project and deleting Derived Data fixed it for me.

**Comment by kballard** [comment]
The Xcode release notes says that the Copy Headers build phase should be before Compile Sources or you may hit this issue. I'm sure having an app to reproduce this would help prove that you've fixed it, but you don't actually need an app to realize that CocoaPods is putting the Compile Sources phase before Copy Headers.

My advice is to change CocoaPods to put Copy Headers before Compile Sources and mark this as done, and then someone can reopen it if they're still seeing the issue with that fix in place.

**Comment by igor-makarov** [comment]
Unfortunately I tried doing that in post, and it only caused the issue to change to a different one. 

Ultimately, I think this one is on XCode.

**Comment by kballard** [comment]
The fact that you got a different issue doesn't mean this one needs to be addressed, it just means there are multiple issues with Xcode 10 beta.

I agree that Xcode should fix this issue themselves. But it's not clear if they *will* fix it by the time Xcode 10 GMs. It's listed as a Known Issue in the release notes, but Apple doesn't always fix all Known Issues.

**Comment by jafara** [comment]
This is an XCode issue.

**Comment by segiddins** [comment]
Please test whether https://github.com/CocoaPods/Xcodeproj/pull/592 fixes the issue for you

**Comment by igor-makarov** [comment]
I have to say that this did not help.
It doesn't hurt either and seems to improve build step order correctness.

**Comment by DominikBucher12** [comment]
I had exact same issue with RxCocoa. Somehow upgrading with`pod repo update`, then reinstalling pods (`pod install`) and then cleaning the whole project build folder solved my issue. I know it's silly comment, but have you tried this? :)

**Comment by segiddins** [comment]
@igor-makarov can you share a project that has this issue, then?

**Comment by igor-makarov** [comment]
Unfortunately no, this happens in our main repo which is private. Also, the failures are intermittent not deterministic.

**Comment by jafara** [comment]
I told you its an XCode issue but everyne downvoted me.

**Comment by kballard** [comment]
@jafara You were downvoted because we already know it's an Xcode issue. We just don't know if Xcode is going to fix the issue before GM, so having CocoaPods be compatible with Xcode 10 beta's current requirements would be beneficial all around.

**Comment by segiddins** [comment]
I’m sorry but it’s incredibly hard to determine if a change we’re making helps or not unless we can test it out. I understand if you can’t share your existing project, but you can also try to reproduce in a clean project.

**Comment by mahgolfa** [comment]
a  (command + k) or ( command + shift + k ) would fix the problem , or at least it did for me

**Comment by gobetti** [comment]
Clean and rebuild works for me too, but I can't waste that time so my favorite solution for now is to switch to the legacy build system

**Comment by GuoZhiQiang** [comment]
Will this help you? 
 - Open your Xcode
 - Click top menu: `File` ---> `Workspace Settings`
 - Then set the `Build System` option to `New Build System`

@hhfa008 @jafara @igor-makarov

**Comment by RScottLewis** [comment]
I had exact same issue with RxAtomic. Somehow upgrading withpod repo update, then reinstalling pods (pod install) and then cleaning the whole project build folder, plus, I opened the project rather than the workspace solved my issue.

**Comment by ryuheechul** [comment]
I had this issue and my cocoapods' version was `1.3.1` so I suspected the old version may causes the issue so I updated `pod`via `$ sudo gem install cocoapods` it got bumped to `1.6.0` and I re-installed my pods via `$ pod install` and I built my project again and that issue was gone.

Hopefully this is helpful.

**Comment by Jerome1210** [comment]
I encounter this issue when I add a share extension and make R.generated.swift file to its target membership. 

Use legacy build fix this issue.

**Comment by GabrielScalici** [comment]
There is a shortcut to fix this issue, working for me.
In the Xcode, press command + shift + K to clean and press play button again.

**Comment by ArnRdy** [comment]
> I encounter this issue when I add a share extension and make R.generated.swift file to its target membership.
> 
> Use legacy build fix this issue.

I faced this issue and was able to fix it by adding scripts (same process used while setting up R.swift) to generate R.generated.swift in the share extension folder. Also, you need to add this file to XCode after it is generated.

**Comment by smity81435** [comment]
`pod clean && pod install`
Was my fix

**Comment by veer-suthar** [comment]
I am facing same issue in Xcode Version 12.4 (12D4e) & cocoapods 1.10.1

Nothing helping.

**Comment by kesha-antonov** [comment]
With `My Mac` target clean doesn't help

**Comment by kavin-zhou** [comment]
I tried pod clean && pod install，but not working for me.  
And command + shift + K is working for me at first, but it reappeared soon after.

**Comment by cody-munger-wm** [comment]
5 years later and it's still not fixed.  Sometimes it compiles but I'm not sure what conditions result in success.  cmd+shift+k doesn't work, switching the build system doesn't work because I also have baked in xcframework.  This does seem to be an obj-c to swift issue.  I'm tempted to ditch the entire pod and include it as a source folder.

**Comment by jgeig001** [comment]
Same problems. Only workaround by clean build (cmd + shift + k) before each build. Everything else did not work.
Fixed by downgrade XCode from 13.3 to 13.2.

**Comment by derb21** [comment]
Don"t do npx pod-install.  actually go to the ios directory and do pod install. Fixed for me.
